Test/Share/Test-GetFileShare.ps1

```diff
@@ -47,7 +47,7 @@ function Test-ShouldGetSpecificShare
 
 function Test-ShouldGetOnlyFileShares
 {
-    $nonFileShares = Get-WmiObject -Class 'Win32_Share' | Where-Object { $_.Type -ne 0 -and $_.Type -ne 2147483648 }
+    $nonFileShares = Get-Cim -Class 'Win32_Share' | Where-Object { $_.Type -ne 0 -and $_.Type -ne 2147483648 }

```

Comment:
The `Where-Object` is an `-and`, so you can change it to two `Where-Object` cmdlets in the pipeline:

```powershell
$nonFileShares = 
    Get-CCimInstance -Class 'Win32_Share' |
    Where-Object 'Type' -NE 0 |
    Where-Object 'Type' -NE 2147483648
```